src/pybind/mgr/dashboard_v2/tests/test_auth.py

```diff
@@ -2,85 +2,103 @@
 
 from __future__ import absolute_import
 
+import json
 import time
+
 from cherrypy.lib.sessions import RamSession
 from cherrypy.test import helper
 from mock import patch
 
-from ..auth import Auth
 from ..module import Module, cherrypy
+from ..tools import load_controller
 
 class Ping(object):
     @cherrypy.expose
     @cherrypy.tools.allow(methods=['POST'])
     def ping(self):
         pass
 
+
 class AuthTest(helper.CPWebCase):
     @staticmethod
     def setup_server():
         module = Module('dashboard', None, None)
-        auth = Auth(module)
-        cherrypy.tools.autenticate = cherrypy.Tool('before_handler', auth.check_auth)
-        cherrypy.tree.mount(auth, "/api/auth")
+        AuthTest.Auth = load_controller(module, 'Auth')
+
+        cherrypy.tools.autenticate = cherrypy.Tool('before_handler',
+                                                   AuthTest.Auth.check_auth)
+
+        cherrypy.tree.mount(AuthTest.Auth(), "/api/auth")
         cherrypy.tree.mount(Ping(), "/api/test",
                             config={'/': {'tools.autenticate.on': True}})
-        module.set_localized_config('session-expire','0.5')
+        module.set_localized_config('session-expire','2')
         module.set_localized_config('username','admin')
-        pass_hash = Auth.password_hash('admin')
+        pass_hash = AuthTest.Auth.password_hash('admin')
         module.set_localized_config('password', pass_hash)
 
+    def _request(self, url, method, data=None):
```

Comment:
I like that. Can you make this generally available?

Reply:
yes, that's the idea. I will do that in another PR